From 2eb1017384ccd23dc3e54c8811c106c41a530c02 Mon Sep 17 00:00:00 2001
From: Kevin Eyssartier <kevin.eyssartier@thalesgroup.com>
Date: Fri, 30 Jun 2023 16:44:21 +0000
Subject: [PATCH] cv32a6, cv64a6 -> cv32a6ima, cv32a6imac, cv64a6gc

A new configuration has appeared !
The differentiation between cv32a6ima to optimize fpga ressources and the historic cv32a6imac.
---
 ..._genesysII.dts => cv32a6ima_genesysII.dts} |   0
 arch/riscv/dts/cv32a6imac_genesysII.dts       | 145 ++++++++++++++++++
 ...6_genesysII.dts => cv64a6gc_genesysII.dts} |   0
 ...openhwgroup_cv32a6ima_genesysII_defconfig} |   2 +-
 ...openhwgroup_cv32a6imac_genesysII_defconfig |  45 ++++++
 ... openhwgroup_cv64a6gc_genesysII_defconfig} |   2 +-
 6 files changed, 192 insertions(+), 2 deletions(-)
 rename arch/riscv/dts/{cv32a6_genesysII.dts => cv32a6ima_genesysII.dts} (100%)
 create mode 100644 arch/riscv/dts/cv32a6imac_genesysII.dts
 rename arch/riscv/dts/{cv64a6_genesysII.dts => cv64a6gc_genesysII.dts} (100%)
 rename configs/{openhwgroup_cv32a6_genesysII_defconfig => openhwgroup_cv32a6ima_genesysII_defconfig} (95%)
 create mode 100644 configs/openhwgroup_cv32a6imac_genesysII_defconfig
 rename configs/{openhwgroup_cv64a6_genesysII_defconfig => openhwgroup_cv64a6gc_genesysII_defconfig} (95%)

diff --git a/arch/riscv/dts/cv32a6_genesysII.dts b/arch/riscv/dts/cv32a6ima_genesysII.dts
similarity index 100%
rename from arch/riscv/dts/cv32a6_genesysII.dts
rename to arch/riscv/dts/cv32a6ima_genesysII.dts
diff --git a/arch/riscv/dts/cv32a6imac_genesysII.dts b/arch/riscv/dts/cv32a6imac_genesysII.dts
new file mode 100644
index 0000000000..54fc2535ea
--- /dev/null
+++ b/arch/riscv/dts/cv32a6imac_genesysII.dts
@@ -0,0 +1,145 @@
+/dts-v1/;
+
+/ {
+  #address-cells = <2>;
+  #size-cells = <2>;
+  compatible = "eth,cva6-bare-dev";
+  model = "eth,cva6-bare";
+  chosen {
+    stdout-path = "/soc/uart@10000000:115200";
+    //tick-timer =  "/cpus/cpu@0";
+  };
+  cpus {
+    #address-cells = <1>;
+    #size-cells = <0>;
+    timebase-frequency = <25000000>; // 25 MHz
+    CPU0: cpu@0 {
+      clock-frequency = <50000000>; // 50 MHz
+      device_type = "cpu";
+      reg = <0>;
+      status = "okay";
+      compatible = "eth, cva6", "riscv";
+      riscv,isa = "rv32imac";
+      mmu-type = "riscv,sv32";
+      tlb-split;
+      // HLIC - hart local interrupt controller
+      CPU0_intc: interrupt-controller {
+        #interrupt-cells = <1>;
+        interrupt-controller;
+        compatible = "riscv,cpu-intc";
+      };
+    };
+  };
+  memory@80000000 {
+    device_type = "memory";
+    reg = <0x0 0x80000000 0x0 0x40000000>;
+  };
+  leds {
+    compatible = "gpio-leds";
+    heartbeat-led {
+      gpios = <&xlnx_gpio 1 0>;
+      linux,default-trigger = "heartbeat";
+      retain-state-suspended;
+    };
+  };
+  L26: soc {
+    #address-cells = <2>;
+    #size-cells = <2>;
+    compatible = "eth,cva6-bare-soc", "simple-bus";
+    ranges;
+    clint@2000000 {
+      compatible = "riscv,clint0";
+      interrupts-extended = <&CPU0_intc 3 &CPU0_intc 7>;
+      reg = <0x0 0x2000000 0x0 0xc0000>;
+      reg-names = "control";
+    };
+    PLIC0: interrupt-controller@c000000 {
+      #address-cells = <0>;
+      #interrupt-cells = <1>;
+      compatible = "riscv,plic0";
+      interrupt-controller;
+      interrupts-extended = <&CPU0_intc 11 &CPU0_intc 9>;
+      reg = <0x0 0xc000000 0x0 0x4000000>;
+      riscv,max-priority = <7>;
+      riscv,ndev = <30>;
+    };
+//  debug-controller@0 {
+//    compatible = "riscv,debug-013";
+//    interrupts-extended = <&CPU0_intc 65535>;
+//    reg = <0x0 0x0 0x0 0x1000>;
+//    reg-names = "control";
+//  };
+    uart@10000000 {
+      compatible = "ns16750", "ns16550";
+      reg = <0x0 0x10000000 0x0 0x1000>;
+      clock-frequency = <50000000>;
+      current-speed = <115200>;
+      interrupt-parent = <&PLIC0>;
+      interrupts = <1>;
+      reg-shift = <2>; // regs are spaced on 32 bit boundary
+      reg-io-width = <4>; // only 32-bit access are supported
+    };
+    timer@18000000 {
+      compatible = "pulp,apb_timer";
+      interrupts = <0x00000004 0x00000005 0x00000006 0x00000007>;
+      reg = <0x00000000 0x18000000 0x00000000 0x00001000>;
+      interrupt-parent = <&PLIC0>;
+      reg-names = "control";
+    };
+    xps-spi@20000000 {
+      compatible = "xlnx,xps-spi-2.00.b", "xlnx,xps-spi-2.00.a";
+      #address-cells = <1>;
+      #size-cells = <0>;
+      interrupt-parent = <&PLIC0>;
+      interrupts = < 2 2 >;
+      reg = < 0x0 0x20000000 0x0 0x1000 >;
+      fifo-size = <256>;
+      xlnx,family = "kintex7";
+      xlnx,fifo-exist = <0x1>;
+      xlnx,num-ss-bits = <0x1>;
+      xlnx,num-transfer-bits = <0x8>;
+      xlnx,sck-ratio = <0x4>;
+
+      mmc@0 {
+        compatible = "mmc-spi-slot";
+        reg = <0>;
+        spi-max-frequency = <12500000>;
+        voltage-ranges = <3300 3300>;
+        disable-wp;
+      };
+
+      // mmc-slot@0 {
+      //   compatible = "fsl,mpc8323rdb-mmc-slot", "mmc-spi-slot";
+      //   reg = <0>;  //Chip select 0
+      //   spi-max-frequency = <12500000>;
+      //   voltage-ranges = <3300 3300>;
+      //   //interrupts = < 2 2 >;
+      //   //interrupt-parent = <&PLIC0>;
+      // };
+    };
+    eth: lowrisc-eth@30000000 {
+      compatible = "lowrisc-eth";
+      device_type = "network";
+      interrupt-parent = <&PLIC0>;
+      interrupts = <3 0>;
+      local-mac-address = [00 18 3e 02 e3 7f]; // This needs to change if more than one GenesysII on a VLAN
+      reg = <0x0 0x30000000 0x0 0x8000>;
+    };
+    xlnx_gpio: gpio@40000000 {
+      #gpio-cells = <2>;
+      compatible = "xlnx,xps-gpio-1.00.a";
+      gpio-controller ;
+      reg = <0x0 0x40000000 0x0 0x10000 >;
+      xlnx,all-inputs = <0x0>;
+      xlnx,all-inputs-2 = <0x0>;
+      xlnx,dout-default = <0x0>;
+      xlnx,dout-default-2 = <0x0>;
+      xlnx,gpio-width = <0x8>;
+      xlnx,gpio2-width = <0x8>;
+      xlnx,interrupt-present = <0x0>;
+      xlnx,is-dual = <0x1>;
+      xlnx,tri-default = <0xffffffff>;
+      xlnx,tri-default-2 = <0xffffffff>;
+    };
+  };
+};
diff --git a/arch/riscv/dts/cv64a6_genesysII.dts b/arch/riscv/dts/cv64a6gc_genesysII.dts
similarity index 100%
rename from arch/riscv/dts/cv64a6_genesysII.dts
rename to arch/riscv/dts/cv64a6gc_genesysII.dts
diff --git a/configs/openhwgroup_cv32a6_genesysII_defconfig b/configs/openhwgroup_cv32a6ima_genesysII_defconfig
similarity index 95%
rename from configs/openhwgroup_cv32a6_genesysII_defconfig
rename to configs/openhwgroup_cv32a6ima_genesysII_defconfig
index 764b0a2849..41814174ba 100644
--- a/configs/openhwgroup_cv32a6_genesysII_defconfig
+++ b/configs/openhwgroup_cv32a6ima_genesysII_defconfig
@@ -1,5 +1,5 @@
 CONFIG_RISCV=y
-CONFIG_DEFAULT_DEVICE_TREE="cv32a6_genesysII"
+CONFIG_DEFAULT_DEVICE_TREE="cv32a6ima_genesysII"
 CONFIG_TARGET_OPENHWGROUP_CVA6_GENESYSII=y
 CONFIG_RISCV_SMODE=y
 # CONFIG_RISCV_ISA_C is not set
diff --git a/configs/openhwgroup_cv32a6imac_genesysII_defconfig b/configs/openhwgroup_cv32a6imac_genesysII_defconfig
new file mode 100644
index 0000000000..a32127c82f
--- /dev/null
+++ b/configs/openhwgroup_cv32a6imac_genesysII_defconfig
@@ -0,0 +1,45 @@
+CONFIG_RISCV=y
+CONFIG_DEFAULT_DEVICE_TREE="cv32a6imac_genesysII"
+CONFIG_TARGET_OPENHWGROUP_CVA6_GENESYSII=y
+CONFIG_RISCV_SMODE=y
+# CONFIG_RISCV_ISA_C is not set
+CONFIG_OF_BOARD_SETUP=y
+CONFIG_BOOTDELAY=5
+CONFIG_USE_BOOTCOMMAND=y
+CONFIG_BOOTCOMMAND="mmc info; mmc read 90000000 100000 7000; setenv fdt_high 0xffffffff; bootm 90000000 - $(fdtcontroladdr)"
+CONFIG_DISPLAY_CPUINFO=y
+CONFIG_SPL_SPI_FLASH_MTD=y
+CONFIG_CMD_GPT=y
+# CONFIG_RANDOM_UUID is not set
+CONFIG_CMD_MMC=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MDIO=y
+CONFIG_CMD_PING=y
+CONFIG_OF_EMBED=y
+CONFIG_MMC=y
+# CONFIG_MMC_WRITE is not set
+CONFIG_MMC_SPI=y
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_SPI_FLASH_ATMEL=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_SST=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_SPI_FLASH_XMC=y
+CONFIG_PHY_REALTEK=y
+CONFIG_LOWRISC_DIGILENT_100MHZ=y
+# CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
+CONFIG_SPI_FLASH_MTD=y
+CONFIG_SYS_NS16550=y
+CONFIG_SPI=y
+CONFIG_XILINX_SPI=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_FAT=y
+CONFIG_FIT=y
diff --git a/configs/openhwgroup_cv64a6_genesysII_defconfig b/configs/openhwgroup_cv64a6gc_genesysII_defconfig
similarity index 95%
rename from configs/openhwgroup_cv64a6_genesysII_defconfig
rename to configs/openhwgroup_cv64a6gc_genesysII_defconfig
index 7e8dbe2d51..2f627fd3b1 100644
--- a/configs/openhwgroup_cv64a6_genesysII_defconfig
+++ b/configs/openhwgroup_cv64a6gc_genesysII_defconfig
@@ -1,5 +1,5 @@
 CONFIG_RISCV=y
-CONFIG_DEFAULT_DEVICE_TREE="cv64a6_genesysII"
+CONFIG_DEFAULT_DEVICE_TREE="cv64a6gc_genesysII"
 CONFIG_TARGET_OPENHWGROUP_CVA6_GENESYSII=y
 CONFIG_ARCH_RV64I=y
 CONFIG_RISCV_SMODE=y
